impeller/display_list/dl_dispatcher.cc

```diff
@@ -484,12 +488,24 @@ static std::shared_ptr<ColorFilter> ToColorFilter(
 // |flutter::DlOpReceiver|
 void DlDispatcher::setColorFilter(const flutter::DlColorFilter* filter) {
   // Needs https://github.com/flutter/flutter/issues/95434
-  paint_.color_filter = ToColorFilter(filter);
+  if (paint_.color_filter) {
+    auto color_filter = ToColorFilter(filter);
+    paint_.color_filter = std::make_shared<MergedColorFilter>(
+        std::move(paint_.color_filter), color_filter);
+  } else {
+    paint_.color_filter = ToColorFilter(filter);
+  }
 }
 
 // |flutter::DlOpReceiver|
 void DlDispatcher::setInvertColors(bool invert) {
-  paint_.invert_colors = invert;
+  if (paint_.color_filter) {
+    auto invert_filter = ColorFilter::MakeMatrix(kColorInversion);
+    paint_.color_filter = std::make_shared<MergedColorFilter>(
+        invert_filter, std::move(paint_.color_filter));
+  } else {
+    paint_.color_filter = ColorFilter::MakeMatrix(kColorInversion);
+  }
```

Comment:
This sucks but there is no good answer to:

1. Be resilient to dl dispatch order and
2. Keep paint as a struct with no private fields.

Reply:
To solve the composing problem, one possibility would be to add a "GetInverted()" to `ColorFilter` that returns a new composed filter with the color filter inversion appended.

Reply:
(This is one of those cases where I would clone things liberally to prevent bugs -- the only reason these things are smart pointers is for dynamic dispatch, but like so many things in Impeller, these should really just be discriminated unions, but we currently have an aversion for those in C++)

Reply:
Right, this is 2.5 which is "callers need to be aware of both fields" which I guess is just us, but not idea.